pkg/storage/stores/series/series_index_store.go

```diff
@@ -293,14 +294,21 @@ func (c *indexReaderWriter) chunksToSeries(ctx context.Context, in []logproto.Ch
 		numConcurrent,
 		func(_ context.Context, idx int) error {
 			res, err := jobs[idx]()
-			if res != nil {
-				results = append(results, res...)
+			if err != nil {
+				return err
 			}
-			return err
+			resultsChan <- res
```

Comment:
I am unable to recreate this bug locally (despite running with a count to 10k), but I'm curious if we just assign the results using `results[idx]` instead of introducing an additional concurrency data structure here, would that also fix the problem? I can see the race condition with reassigning `results`, but only assigning to specific indexes shouldn't have the same problem?